app/main.py

```diff
@@ -1,34 +1,215 @@
+                self.field[(deck.row, deck.column)] = ship
+
+    def print_field(self) -> None:
+        whole_field = dict()
+        for row_index in range(10):
+            for column_index in range(10):
+                symbol = "~"
+                if (row_index, column_index) in self.field:
+                    symbol = "□"
+                whole_field[(row_index, column_index)] = symbol
+
+        for location in whole_field:
+            ship = self.field.get(location)
+            if ship:
+                symbol = "*"
+                if ship.is_drowned:
+                    symbol = "x"
+                for deck in ship.decks:
+                    if not deck.is_alive:
+                        whole_field[(deck.row, deck.column)] = symbol
+
+        for index, location in enumerate(whole_field):
+            if index % 10 == 0:
+                print("")
+            print(f"{whole_field.get(location)}      ", end="")
+
+    def _validate_field(self) -> bool:
+        count_ships = dict()
+
+        ships = set(self.field.values())
+        for ship in ships:
+            ship_length = len(ship.decks)
+
+            if count_ships.get(ship_length):
+                count_ships[ship_length] += 1
+                continue
+            count_ships[ship_length] = 1
+
+        checks = (
+            count_ships.get(1) == 4,
+            count_ships.get(2) == 3,
+            count_ships.get(3) == 2,
+            count_ships.get(4) == 1,
+            self.check_neighbours_cells(ships)
+        )
+
+        if all(checks):
+            return True
+        return False
+
+    def check_neighbours_cells(self, ships: set[Ship]) -> bool:
+        for ship in ships:
+            locations = []
+            for deck in ship.decks:
+                locations.append((deck.row, deck.column))
+
+            for location in locations:
+                combinations = [
+                    (0, 1),
+                    (0, -1),
+                    (-1, -1),
+                    (-1, 1),
+                    (1, 1),
+                    (1, -1),
+                    (1, 0),
+                    (-1, 0),
+                ]
+                not_allowed_cells = itertools.filterfalse(
+                    lambda row_and_column: self.check_cell(
+                        locations,
+                        location,
+                        row_and_column
+                    ),
+                    iter(combinations)
+                )
+
+                for _ in not_allowed_cells:
+                    return False
+        return True
+
+    def check_cell(
+            self,
+            locations: list[tuple],
+            location: tuple,
+            row_and_column: tuple
+    ) -> bool:
+        curr_location = (
+            location[0] + row_and_column[0],
+            location[1] + row_and_column[1]
+        )
+
+        checks = curr_location in self.field, curr_location not in locations
+        return not all(checks)
```

Comment:
Place your checks inside `all`, there is no need for a separate variable here as well.